velox/connectors/hive/iceberg/tests/TransformE2ETest.cpp

```diff
@@ -0,0 +1,646 @@
+
+  verifyPartitionCount(
+      outputDirectory->getPath(), kDefaultNumBatches * rowsPerBatch);
+
+  const auto& partitionDirs = firstLevelDirectories(outputDirectory->getPath());
+  for (const auto& dir : partitionDirs) {
+    const auto& dirName = std::filesystem::path(dir).filename().string();
+    verifyPartitionData(rowType, dir, dirName, 1);
+  }
+}
+
+TEST_F(TransformE2ETest, partitionNamingConventions) {
+  std::string binaryData("binary\0data\1\2\3", 14);
+  auto rowVector = makeRowVector(
+      {
+          "c_int",
+          "c_bigint",
+          "c_varchar",
+          "c_varchar2",
+          "c_decimal",
+          "c_varbinary",
+      },
+      {
+          makeConstant(42, 1, INTEGER()),
+          makeConstant(static_cast<int64_t>(9'876'543'210), 1, BIGINT()),
+          makeConstant("test string", 1, VARCHAR()),
+          makeNullConstant(TypeKind::VARCHAR, 1),
+          makeConstant(static_cast<int64_t>(1'234'567'890), 1, DECIMAL(18, 3)),
+          makeConstant(StringView(binaryData), 1, VARBINARY()),
+      });
+
+  std::vector<test::PartitionField> partitionTransforms = {
+      {0, TransformType::kIdentity, std::nullopt}, // c_int.
+      {1, TransformType::kIdentity, std::nullopt}, // c_bigint.
+      {2, TransformType::kIdentity, std::nullopt}, // c_varchar.
+      {4, TransformType::kIdentity, std::nullopt}, // c_decimal.
+      {5, TransformType::kIdentity, std::nullopt}, // c_varbinary.
+      {3, TransformType::kIdentity, std::nullopt}, // c_varchar2.
+  };
+
+  auto outputDirectory = writeBatchesWithTransforms(
+      {rowVector}, asRowType(rowVector->type()), partitionTransforms);
+
+  const auto& actualPartitionNames =
+      listDirectoriesRecursively(outputDirectory->getPath());
+
+  // Build expected partition folder names.
+  std::vector<std::string> expectedPartitionNames = {
+      "c_int=42",
+      "c_bigint=9876543210",
+      "c_varchar=test+string",
+      "c_decimal=1234567.890",
+      "c_varbinary=YmluYXJ5AGRhdGEBAgM%3D",
+      "c_varchar2=null",
+  };
+
+  ASSERT_EQ(actualPartitionNames, expectedPartitionNames)
+      << "Partition folder names do not match expected values";
+}
+
+TEST_F(TransformE2ETest, bucket) {
+  constexpr int32_t numBuckets = 4;
+  auto rowType = ROW({"c_varchar"}, {VARCHAR()});
+  auto vectors =
+      createTestData(rowType, kDefaultNumBatches, kDefaultRowsPerBatch);
+
+  auto outputDirectory = writeBatchesWithTransforms(
+      vectors, rowType, {{0, TransformType::kBucket, numBuckets}});
+
+  const auto& partitionDirs = firstLevelDirectories(outputDirectory->getPath());
+  ASSERT_LE(partitionDirs.size(), numBuckets);
+
+  int32_t totalRowsInPartitions = 0;
+  for (const auto& dir : partitionDirs) {
+    auto splits = createSplitsForDirectory(dir);
+    auto countPlan = PlanBuilder().tableScan(rowType).planNode();
+    auto partitionRowCount =
+        AssertQueryBuilder(countPlan).splits(splits).runWithoutResults();
```

Comment:
Let's rename runWithoutResults to countResults. That would be more consistent with copyResults.